presto-cli/src/main/java/io/prestosql/cli/Console.java

```diff
@@ -170,7 +170,13 @@ private String getPassword()
 
         java.io.Console console = System.console();
         if (console == null) {
-            throw new RuntimeException("No console from which to read password");
+            try {
+                jline.console.ConsoleReader consoleReader = new jline.console.ConsoleReader(System.in, System.err);
+                return consoleReader.readLine("Password: ", Character.valueOf((char) 0));
+            }
+            catch (IOException exception) {
+                throw new RuntimeException("No console from which to read password");
```

Comment:
Throw as `UncheckedIOException`. Also, this error might not be correct anymore. Probably better to make it more general: `Failed to read password from console`

Reply:
Applied